robyn/__init__.py

```diff
@@ -369,9 +396,7 @@ def get(
         """
 
         def inner(handler):
-            self.openapi.add_openapi_path_obj("get", endpoint, openapi_name, openapi_tags, handler)
-
-            return self.add_route(HttpMethod.GET, endpoint, handler, const, auth_required)
```

Comment:
Hey @dave42w 👋 

Could you explain this modification a bit?

Reply:
Sure.

My goal is that we do the minimum processing of routes while in the app setup stage. ie between `app = Robyn(__file__)` and `app.start(...)`

My logic is that things can happen in quite a dynamic way in this section of the code, developers might do things in an odd order (include a subrouter and then add more routes to it). Or include a subrouter in a subrouter and then include the subrouter and all it's subsubrouters. Or add middleware before the routes or in the middle of them. This could happen because we bring in a subrouter via a module and there it includes routes and middleware. Or want different middleware for the subrouter from accounting compared to sales etc.

Therefore if we call `add_openapi_path_obj` in add_route the path might not yet be fixed. So it takes more logic (and potentially doing things multiple times) to handle these variations.

When we have calls to openapi scattered around the code then it gets expensive in runtime and maintenance if in every case we want to only make the call if openapi is enabled. It gets slow to maintain if we have 10 calls to `add_openapi_path_obj` and need to change it.

Of course if we are going to collect together all the calls to `add_openapi_path_obj` and do them in one place as the server starts then we need to have the handler, openapi_name and openapi_tags available during `app.start()`. This is why I have added them to the Route class.

I'm pretty sure that when I refactor add_route it will be a lot cleaner if we create an incomplete Route instance much earlier and then pass the single object around the system rather than 7 arguments. Deep in the system we do the work to complete function_info to finish the Route instance. That will also reduce the need for parsing the arguments all over the place (maybe Route needs to be a dataclass to validate itself). All that is for a future PR.